### PR TITLE
sci-physics/vgm: Version bump to 4.5, fixes for 17.1 profile

### PR DESCRIPTION
**Pull request by olifre**
This series contains:
- A patch for finding CLHEP needed with the 17.1 profile (upstreamed) and adapting dependencies to the `c++11` USE-flag of `sci-physics/root`. 
- A version bump to 4.5.

### CODE REVIEW

File: sci-physics/vgm/vgm-4.4.ebuild [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
```
Comment (mgorny) [inline sci-physics/vgm/vgm-4.4.ebuild]: Similarly to your other PR, this looks like a case for revbump.

Comment (olifre) [reply sci-physics/vgm/vgm-4.4.ebuild]: Ok, done! 
I have added the patch which fixes (re)build for those on 17.1 also to the non-revbumped ebuild, so the revbump is for the EAPI bump alone.

Comment (mgorny) [reply sci-physics/vgm/vgm-4.4.ebuild]: Just leave the old ebuild alone, you are doing a version bump anyway.

Comment (olifre) [reply sci-physics/vgm/vgm-4.4.ebuild]: Ok, done!

**Comment by mgorny** [comment]
```
FAILED: test/CMakeFiles/vgm_test.dir/src/TstDetectorConstruction.cc.o 
/usr/lib/ccache/bin/x86_64-pc-linux-gnu-g++-8.3.0 -DG4INTY_USE_XT -DG4LIB_BUILD_DLL -DG4MULTITHREADED -DG4UI_USE -DG4UI_USE_TCSH -DG4VERBOSE -DG4VIS_USE -DG4VIS_USE_OPENGL -DG4VIS_USE_OPENGLX -DG4_STORE_TRAJECTORY -DVGM_WITH_G4ROOT -I/usr/lib64/CLHEP-2.4.1.2/../../include -I/usr/lib/root/6.18/include -I/tmp/portage/sci-physics/vgm-4.5/work/vgm/packages/VGM/include -I/tmp/portage/sci-physics/vgm-4.5/work/vgm/packages/ClhepVGM/include -I/tmp/portage/sci-physics/vgm-4.5/work/vgm/packages/BaseVGM/include -I/tmp/portage/sci-physics/vgm-4.5/work/vgm/packages/XmlVGM/include -I/tmp/portage/sci-physics/vgm-4.5/work/vgm/packages/RootGM/include -I/tmp/portage/sci-physics/vgm-4.5/work/vgm/packages/Geant4GM/include -I/tmp/portage/sci-physics/vgm-4.5/work/vgm/test/include -I/usr/include/g4root -isystem /usr/include/Geant4  -DNDEBUG -march=k8-sse3 -mcx16 -O2 -pipe -W -Wall -pedantic -Wno-non-virtual-dtor -Wno-long-long -Wwrite-strings -Wpointer-arith -Woverloaded-virtual -Wno-variadic-macros -Wshadow -pipe -DG4USE_STD11 -pthread -ftls-model=global-dynamic -std=c++11 -std=c++11    -std=c++11 -MD -MT test/CMakeFiles/vgm_test.dir/src/TstDetectorConstruction.cc.o -MF test/CMakeFiles/vgm_test.dir/src/TstDetectorConstruction.cc.o.d -o test/CMakeFiles/vgm_test.dir/src/TstDetectorConstruction.cc.o -c /tmp/portage/sci-physics/vgm-4.5/work/vgm/test/src/TstDetectorConstruction.cc
distcc[965] ERROR: compile (null) on localhost failed
/tmp/portage/sci-physics/vgm-4.5/work/vgm/test/src/TstDetectorConstruction.cc:27:10: fatal error: G4GDMLParser.hh: No such file or directory
 #include "G4GDMLParser.hh"
          ^~~~~~~~~~~~~~~~~
```

[sci-physics:vgm-4.5:20190718-161338.log](https://github.com/gentoo/gentoo/files/3407433/sci-physics.vgm-4.5.20190718-161338.log)

**Comment by olifre** [comment]
Really sorry for that :cry: . Apparently I should disable all default USE flags on dependencies and not believe in upstream's stated dependencies to catch this earlier. 

This was caused by missing GDML flag on Geant (`sci-physics/geant[gdml]`), should be fixed now.

**Pull request closed without merging**